skyportal/tests/api/test_annotations_on_photometry.py

```diff
@@ -0,0 +1,420 @@
+    # instead, try updating the existing annotation if you have new information!
+    status, data = api(
+        'POST',
+        f'photometry/{photometry_id}/annotations',
+        data={
+            'origin': 'kowalski',
+            'data': {'offset_from_host_galaxy': 1.5},
+            'group_ids': [public_group.id],
+        },
+        token=annotation_token,
+    )
+
+    assert status in [500, 400]
+    assert 'duplicate key value violates unique constraint' in data["message"]
+
+    status, data = api(
+        'GET',
+        f'photometry/{photometry_id}/annotations/{annotation_id}',
+        token=annotation_token,
+    )
+
+    assert status == 200
+    assert data['data']['data']['offset_from_host_galaxy'] == 1.5
+
+
+def test_add_and_retrieve_annotation_group_access(
+    annotation_token_two_groups,
+    upload_data_token_two_groups,
+    public_source_two_groups,
+    public_group2,
+    public_group,
+    annotation_token,
+    ztf_camera,
+):
+
+    status, data = api(
+        'POST',
+        'photometry',
+        data={
+            'obj_id': str(public_source_two_groups.id),
+            'mjd': 58000.0,
+            'instrument_id': ztf_camera.id,
+            'flux': 12.24,
+            'fluxerr': 0.031,
+            'zp': 25.0,
+            'magsys': 'ab',
+            'filter': 'ztfg',
+            'group_ids': [public_group2.id],
+            'altdata': {'some_key': 'some_value'},
+        },
+        token=upload_data_token_two_groups,
+    )
+    assert_api(status, data)
+
+    photometry_id = data['data']['ids'][0]
+
+    status, data = api(
+        'POST',
+        f'photometry/{photometry_id}/annotations',
+        data={
+            'origin': 'IPAC',
+            'photometry_id': photometry_id,
+            'data': {'distance_from_host': 7.4},
+            'group_ids': [public_group2.id],
+        },
+        token=annotation_token_two_groups,
+    )
+    assert status == 200
+    annotation_id = data['data']['annotation_id']
+
+    # This token belongs to public_group2
+    status, data = api(
+        'GET',
+        f'photometry/{photometry_id}/annotations/{annotation_id}',
+        token=annotation_token_two_groups,
+    )
+    assert status == 200
+    assert data['data']['data']['distance_from_host'] == 7.4
+
+    # This token does not belong to public_group2
+    status, data = api(
+        'GET',
+        f'photometry/{photometry_id}/annotations/{annotation_id}',
+        token=annotation_token,
+    )
+    assert status == 403
+
+    # Both tokens should be able to view this annotation, but not the underlying photometry
+    status, data = api(
+        'POST',
+        f'photometry/{photometry_id}/annotations',
+        data={
+            'origin': 'kowalski',
+            'photometry_id': photometry_id,
+            'data': {'ACAI_class': 'type Ia'},
+            'group_ids': [public_group.id, public_group2.id],
+        },
+        token=annotation_token_two_groups,
+    )
```

Comment:
I assume this is the same behavior as we had for spectra. Do we want to allow users to view an annotation without being able to view the photometry point? This is a little problematic... we generally don't even let a user know if a photometry ID exists if they don't have permission to view it.

Reply:
So right now photometry read implies that you have obj and photometry available for read. Do you have a more particular concern?

Reply:
I'm just worried that if you can read an annotation on a photometry point that is inaccessible to you, you will get information that the photometry point exists.

Reply:
But this is impossible because the permissions for read on annotation implies a permission on photometry.

Reply:
I am confused because of the code comment above: `Both tokens should be able to view this annotation, but not the underlying photometry`